src/binary-reader.cc

```diff
@@ -1605,6 +1605,9 @@ static void read_custom_section(Context* ctx, uint32_t section_size) {
       uint32_t subsection_size;
       in_u32_leb128(ctx, &name_type, "name type");
       in_u32_leb128(ctx, &subsection_size, "subsection size");
+      size_t subsection_end = ctx->offset + subsection_size;
+      if (subsection_end > ctx->read_end)
+        RAISE_ERROR("invalid sub-section size: extends past end");
```

Comment:
Its a good question, for custom sections that the binary-reader knows about, how should it handle errors.   My feeling is that if we are going to parse a section we should validate its contents.  Perhaps a flag to disable all-parsing of custom sections if people don't what this?  But perhaps we should add this if and when somebody needs it.